.github/workflows/preflight.yml

```diff
@@ -0,0 +1,42 @@
+name: Preflight Tests
+
+on:
+  schedule:
+    - cron: '21 17 * * *'
+  workflow_dispatch:
+    inputs:
+      reason:
+        description: Brief reason for running this workflow manually
+        required: false
+        default: User initiated run
+        type: string
+  push:
+    branches: [master, ci-preflight]
```

Comment:
reminder to self: remove the ci-preflight branch from here before merging the pr